upup/pkg/fi/cloudup/awstasks/sshkey.go

```diff
@@ -176,6 +187,9 @@ func (_ *SSHKey) RenderTerraform(t *terraform.TerraformTarget, a, e, changes *SS
 }
 
 func (e *SSHKey) TerraformLink() *terraform.Literal {
+	if e.KeyFingerprint != nil && e.PublicKey == nil {
```

Comment:
Does it suffice to just do `if e.PublicKey == nil {`  here?

One thing that might make it clearer is to explicitly have a method `func (e *SSHKey) IsExistingKey() bool { return e.PublicKey == nil }` or maybe even a field which you would set in the model.

Reply:
Done